backend/src/cache/server/mutation.go

```diff
@@ -248,7 +252,15 @@ func prepareInitContainer(pod *corev1.Pod, logger *zap.SugaredLogger) ([]corev1.
 
 func prepareMainContainer(pod *corev1.Pod, result string, logger *zap.SugaredLogger) ([]corev1.Container, error) {
 	logger.Infof("Start to prepare dummy containers.")
-	dummyContainers := []corev1.Container{}
+	image := "gcr.io/google-containers/busybox"
+	dummyContainer := corev1.Container{
```

Comment:
Can we not add this extra container here? Here you are running two containers even though it's a cached task.

Maybe you can prepend the echo commands to the `replacedArg`  below. This way we only need to run one dummy container.

Reply:
@drewbutlerbb4 please also update this test case with your new changes. Thanks
https://github.com/kubeflow/kfp-tekton/blob/master/backend/src/cache/server/mutation_test.go#L54-L63

Reply:
`make run-cacheserver-unittests` will verify the cache server tests for you.